core/src/main/resources/reference.conf

```diff
@@ -568,48 +566,35 @@ cassandra-query-journal {
+  # yyyyMMddTHH:mm
+  # Will be rounded down to the start of whatever time bucket it falls into
   # When NoOffset is used it will look for events from this day and forward.
-  first-time-bucket = "20151120"
-
-  # The returned event stream is ordered by the offset (timestamp), which corresponds
-  # to the same order as the write journal stored the events, with inaccuracy due to clock skew
-  # between different nodes. The same stream elements (in same order) are returned for multiple
-  # executions of the query on a best effort basis. The query is using a Cassandra Materialized
-  # View for the query and that is eventually consistent, so different queries may see different
-  # events for the latest events, but eventually the result will be ordered by timestamp
-  # (Cassandra timeuuid column). To compensate for the the eventual consistency the query is
-  # delayed to not read the latest events, the duration of this delay is defined by this 
-  # configuration property.
-  # However, this is only best effort and in case of network partitions
-  # or other things that may delay the updates of the Materialized View the events may be
-  # delivered in different order (not strictly by their timestamp).  
-  eventual-consistency-delay = 10s
-
-  # If you use the same tag for all events for a `persistenceId` it is possible to get
-  # a more strict delivery order than otherwise. This can be useful when all events of
-  # a PersistentActor class (all events of all instances of that PersistentActor class)
-  # are tagged with the same tag. Then the events for each `persistenceId` can be delivered
-  # strictly by sequence number. If a sequence number is missing the query is delayed up 
-  # to the configured `delayed-event-timeout` and if the expected event is still not 
-  # found the stream is completed with failure. This means that there must not be any 
-  # holes in the sequence numbers for a given tag, i.e. all events must be tagged
-  # with the same tag. Set this property to for example 30s to enable this feature.
-  # It is disabled by default.
-  delayed-event-timeout = 0s
+  first-time-bucket = "20151120T00:00"
+
+  # How long to look for delayed events
+  # This works by adding an additional (internal) sequence number to each tag / persistence id
+  # event stream so that the read side can detect missing events. When a gap is detected no new events
+  # are emitted from the stream until either the missing events are found or the timeout is reached
+  # If the event is not found it is checked every `refresh-interval` so do not set this lower than that
+  # if you want at least one retry
+  # When looking for missing events only the current time bucket and the previous bucket are checked meaning that if
+  # clocks are out of sync, or cassandra replication is out by more than your bucket size (minute, hour or day)
+  # then the missing events won't be found
+  events-by-tag-gap-timeout = 10s
```

Comment:
are there any drawbacks of increasing the default somewhat to avoid spurious stream failure? e.g. 30s

Reply:
Only when we see a new persistenceId and its sequence nr is not 1 and the previous events are before the offset for the query. We will look for 30s for things we have no chance of finding

@patriknw  WDYT about a different timeout for this case? I think it should be much smaller.

Reply:
do you mean that it will continue scanning the previous/current bucket during this time until it accepts it as the first? with an new timeout for each persistenceId?

what does that mean for queries that start from an offset? all events will be for an unseen persistenceId initially and they will have seqNr > 1

Reply:
Yes the current stage will keep looking for the missing sequence numbers 1 - N where N is the first sequence number it found for a new persistenceId.

When the user starts at an offset most persistenceIds will not start from 1 so we won't know if they are missing or before the offset.

I was thinking of having a much smaller timeout for this case, maybe even just check the current and last bucket once?